triage/experiments/validate.py

```diff
@@ -0,0 +1,221 @@
+import logging
+import architect
+from architect.utils import convert_str_to_relativedelta
+from triage.validation_primitives import table_should_have_data,\
+    column_should_be_intlike,\
+    column_should_be_booleanlike,\
+    column_should_be_stringlike,\
+    column_should_be_timelike
+
+
+class Validator(object):
+    def __init__(self, db_engine):
+        self.db_engine = db_engine
+
+
+class FeatureAggregationsValidator(Validator):
+    def _validate_keys(self, aggregation_config):
+        for key in ['from_obj', 'intervals', 'groups', 'knowledge_date_column', 'prefix']:
+            if key not in aggregation_config:
+                raise ValueError(
+                    '{} required as key: aggregation config: {}'
+                    .format(key, aggregation_config)
+                )
+
+    def _validate_aggregates(self, aggregation_config):
+        if 'aggregates' not in aggregation_config \
+                and 'categoricals' not in aggregation_config \
+                and 'array_categoricals' not in aggregation_config:
+            raise ValueError(
+                'Need either aggregates, categoricals, or array_categoricals' +
+                ' in {}'.format(aggregation_config)
+            )
+
+    def _validate_categoricals(self, categoricals):
+        conn = self.db_engine.connect()
+        for categorical in categoricals:
+            if 'choice_query' in categorical:
```

Comment:
If `choice_query` isn't specified should we validate that `choices` is (and, likewise, that one and only one of those is given)?